### DISCUSSION TITLE
Source generator for inheritance of blueprint classes

### DISCUSSION THREAD

**Discussion by ScottKane** [nxrighthere/UnrealCLR#58] [Ideas]
Just a thought, couldn't a source generator be used targeting classes marked with an attribute e.g [BlueprintInheritable] to expose managed classes allowing them to be inherited from blueprint. I've seen that it's been mentioned as out of scope but just thinking that not being able to do this will restrict anyone using UnrealCLR from using a LOT of marketplace assets if they for example have a BP inheriting from Character.

**Comment by nxrighthere** [comment]
You still can access base classes without any problems and extend blueprints with managed functions. I'm using several 3rd-party blueprint assets myself mixed with managed code and the class hierarchy of the engine. I only want to improve interop with managed functionality that exposes to blueprints.

Also, keep in mind that this project is one-man effort, so I'm trying to minimize maintenance costs which are: time, energy, nerves, and so on. Nobody is paying me for that. I also learned from other projects that integrates scripting to Unreal, and I don't want to end up in nowhere because of overcomplexity.

**Reply by celeronpm** [reply]
Imagine if Epic allowed you to have a fair monetization strategy so you could actually be supported to grow UnrealCLR!
